src/XMLimport.cpp

```diff
@@ -447,12 +448,36 @@ void XMLimport::readRooms(QMultiHash<int, int>& areaRoomsHash)
             } else {
                 readUnknownMapElement();
             }
-        } else if (isEndElement()) {
+        } else if (isEndElement() && name() == qsl("rooms")) {
             break;
         }
     }
 }
 
+void XMLimport::readRoomFeatures(TRoom* pR)
+{
+    while (!atEnd()) {
+        readNext();
+
+        if (Q_LIKELY(isStartElement())) {
+            if (name() == qsl("features")) {
+                continue;
+            } else if (Q_LIKELY(name() == qsl("feature"))) {
+                readRoomFeature(pR);
+            }
+        } else if (isEndElement() && name() == qsl("features")) {
+            break;
+        }
+    }
+}
+
+void XMLimport::readRoomFeature(TRoom* pR)
+{
+    if (Q_LIKELY(attributes().hasAttribute(qsl("type")))) {
+        pR->userData.insert(qsl("feature_%1").arg(attributes().value(qsl("type"))), qsl("true"));
```

Comment:
IRE mapper uses `feature-` to lets align with that 
```suggestion
        pR->userData.insert(qsl("feature-%1").arg(attributes().value(qsl("type"))), qsl("true"));
```

Reply:
> IRE mapper uses feature- to lets align with that

Whose mapper? The one we ship? Or one developed by @keneanung *et al*? Or something else entirely?

BTW are separate entries, like `feature_shop="true"`, `feature_teleporter="true"` etc. the best way to store the data in the room user data? Or would a `features={"shop"; "teleporter"}` *table* be better, note that the details are being stored as string(s) in the room user data and I am not clear if the latter would automagically get converted to a table when the Lua sub-system reads it or not?

Reply:
> Whose mapper? The one we ship? Or one developed by @keneanung et al? Or something else entirely?

Those are the same :stuck_out_tongue:

We currently use `feature-shop = true` as that makes searching easier, apparently.

Reply:
Okay, then, will make what this PR does the same...